### PR TITLE
Add ELFIO v3.12

### PR DESCRIPTION
**Pull request by Mellich**

### CODE REVIEW

**Comment by vchuravy** [comment]
Why do we need a header only library in Yggdrasil?

**Comment by Mellich** [comment]
As a build dependency. Are there other ways of using header-only libraries without polluting the resulting artifacts?

**Comment by vchuravy** [comment]
Sure you can add it as a second source and not install this header file?

**Comment by giordano** [comment]
We have other header-only packages, to be used as BuildDependency'ies

File: E/ELFIO/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+platforms = [AnyPlatform()]
+
+products = [
+    FileProduct("include/elfio/elfio.hpp", :elfio_hpp)
+]
+
+
+dependencies = []
+
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```
Comment (giordano) [inline E/ELFIO/build_tarballs.jl:26]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")

```

File: E/ELFIO/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+"""
+
+platforms = [AnyPlatform()]
+
+products = [
+    FileProduct("include/elfio/elfio.hpp", :elfio_hpp)
+]
+
+
+dependencies = []
```
Comment (giordano) [inline E/ELFIO/build_tarballs.jl:24]:
```suggestion
dependencies = Dependency[]
```

**Pull request merged**